SimplenoteWidgets/Tools/WidgetDataController.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import SimplenoteFoundation
+import SimplenoteSearch
+
+class WidgetDataController {
+
+    /// Data Controller
+    ///
+    let coreDataManager: CoreDataManager
+
+    /// Notes Controller
+    ///
+    private lazy var notesController = ResultsController<Note>(
+        viewContext: coreDataManager.managedObjectContext,
+        matching: predicateForNotes(),
+        sortedBy: [NSSortDescriptor.descriptorForNotes(sortMode: noteSortMode)]
+    )
+
+    /// Initialization
+    ///
+    init(coreDataManager: CoreDataManager, isPreview: Bool = false) throws {
+        guard !isPreview else {
+            self.coreDataManager = coreDataManager
+            return
+        }
+
+        guard let isLoggedIn = UserDefaults(suiteName: SimplenoteConstants.sharedGroupDomain)?.bool(forKey: .accountIsLoggedIn),
+              isLoggedIn else {
+            throw WidgetError.appConfigurationError
+        }
+
+        self.coreDataManager = coreDataManager
+    }
+
+    /// Sort mode for widgets.  Fetched from main apps sort setting
+    ///
+    private var noteSortMode: SortMode {
+        guard let defaults = UserDefaults(suiteName: SimplenoteConstants.sharedGroupDomain) else {
+            return SortMode.alphabeticallyAscending
+        }
+
+        return SortMode(rawValue: defaults.integer(forKey: .listSortMode))
+            ?? SortMode.alphabeticallyAscending
```

Comment:
I'm starting to think we could maybe use a class called... perhaps WidgetUserDefaults (similar to Options.swift).  Or not.

Reply:
I like it in it's own class.